### PR TITLE
issue65 admin vote supersedes members

### PR DESCRIPTION
**Pull request by leeduan**
Allow admin vote to be used instead of member votes: https://github.com/provenance-io/digital-currency-consortium/issues/65

* Decouples voting and weight by admin from regular members
* Allow admin vote to override need to check member votes

### CODE REVIEW

File: smart-contract/src/state.rs [hunk 1]
```diff
@@ -6,6 +6,7 @@ use cosmwasm_storage::{
     bucket, bucket_read, singleton, singleton_read, Bucket, ReadonlyBucket, ReadonlySingleton,
     Singleton,
 };
+use crate::msg::VoteChoice;
```
Comment (leeduan) [inline smart-contract/src/state.rs:9]: not sure if its best practice to pull in enum from another file vs defining it again and mapping between enums 🤷

File: smart-contract/src/contract.rs [hunk 2]
```diff
@@ -223,28 +224,32 @@ fn try_vote(
...
+                proposal.no += m.weight;
+            }
+        };
 
-    // Add message sender to join request 'voted' vector.
-    proposal.voters.push(info.sender);
+        // Add message sender to join request 'voted' vector.
+        proposal.voters.push(info.sender);
+    } else {
+        // Add admin vote separate from member vote result separate from member votes.
```
Comment (ranadewall) [inline smart-contract/src/contract.rs:250]: nit this comment is confusing to me

Comment (leeduan) [reply smart-contract/src/contract.rs]: haha i forgot to push my earlier change of this, fixed now

**Review by ranadewall — APPROVED** [review]
not sure anyone should trust my code reviews today 🤣

File: smart-contract/src/contract.rs [hunk 3]
```diff
@@ -223,28 +224,32 @@ fn try_vote(
         return Err(contract_err("member has already voted"));
     }
 
-    // Esure member joined before the proposal was created and determine weight.
-    let weight = if let Some(m) = member {
+    if info.sender == state.admin && proposal.admin_vote != None {
```
Comment (scirner22) [inline smart-contract/src/contract.rs:227]: is there a `proposal.admin_vote.is_some()`? I believe that would be more idiomatic.

File: smart-contract/src/state.rs (hunk 1 above)
Comment (scirner22) [reply smart-contract/src/state.rs:9]: The import is fine.

**Review by scirner22 — APPROVED** [review]

File: smart-contract/src/contract.rs (hunk 3 above)
Comment (leeduan) [reply smart-contract/src/contract.rs:227]: ah thanks

**Pull request merged**